sorl/thumbnail/conf/__init__.py

```diff
@@ -1,19 +1,19 @@
 from django.conf import settings as user_settings
-from django.utils.functional import LazyObject
 from sorl.thumbnail.conf import defaults
 
 
-class Settings(object):
-    pass
+class Settings:
+    """
+    Settings proxy that will lookup first in the django settings, and then in the conf
+    defaults.
+    """
+    def __getattr__(self, name):
+        if name != name.upper():
+            raise AttributeError(name)
+        try:
+            return getattr(user_settings, name)
+        except AttributeError:
+            return getattr(defaults, name)
 
 
-class LazySettings(LazyObject):
-    def _setup(self):
-        self._wrapped = Settings()
-        for obj in (defaults, user_settings):
-            for attr in dir(obj):
-                if attr == attr.upper():
-                    setattr(self, attr, getattr(obj, attr))
-
-
-settings = LazySettings()
```

Comment:
Was it necessary to loose the lazy nature of the settings?

I've only had a very cursory look , and it seems possible to keep the lazy nature , but I may have missed a side-effect that would cause.

Reply:
The new implementation is lazy, in the sense that each setting is loaded only when it's value is queried. The former implementation was lazy in the sense that it loaded *all* settings as soon as *one* setting value was queried. The `Settings` object might be loaded multiple times, but that's ok, because creating new objects is free.

So in terms of pure performance, I would argue that the new implementation is more efficient. Also, it is probably more correct as as well, as it takes into account the cases when setting values change (a rare use case, but it happens).